tests/TestAutoType.cpp

```diff
@@ -282,3 +282,30 @@ void TestAutoType::testGlobalAutoTypeRegExp()
     QCOMPARE(m_test->actionChars(), QString("custom_attr_third"));
     m_test->clearActions();
 }
+
+void TestAutoType::testAutoTypeSyntaxChecks()
+{
+    // Huge sequence
+    QCOMPARE(true, AutoType::checkSyntax("{word 23}{F1 23}{~ 23}{% 23}{^}{F12}{(}{) 23}{[}{[}{]}{Delay=23}{+}{-}~+%@fixedstring"));
+
+    QCOMPARE(true, AutoType::checkSyntax("{NUMPAD1 3}"));
+
+    QCOMPARE(true, AutoType::checkSyntax("{BEEP 3 3}"));
+    QCOMPARE(false, AutoType::checkSyntax("{BEEP 3}"));
+
+    QCOMPARE(true, AutoType::checkSyntax("{VKEY 0x01}"));
+    QCOMPARE(true, AutoType::checkSyntax("{VKEY VK_LBUTTON}"));
+    QCOMPARE(true, AutoType::checkSyntax("{VKEY-EX 0x01}"));
+    // Bad sequence
+    QCOMPARE(false, AutoType::checkSyntax("{{{}}{}{}}{{}}"));
```

Comment:
@TheZ3ro While you are at it, would be nice to have `{{}` and `{}}` covered by these tests.

Reply:
Done